components/core/src/clp/clp/FileDecompressor.cpp

```diff
@@ -76,4 +118,125 @@ bool FileDecompressor::decompress_file(
 
     return true;
 }
+
+bool FileDecompressor::decompress_ir(
+        streaming_archive::MetadataDB::FileIterator const& file_metadata_ix,
+        string const& output_dir,
+        string const& temp_output_dir,
+        streaming_archive::reader::Archive& archive_reader,
+        size_t ir_target_size
+) {
+    // Open encoded file
+    auto error_code = archive_reader.open_file(m_encoded_file, file_metadata_ix);
+    if (ErrorCode_Success != error_code) {
+        if (ErrorCode_errno == error_code) {
+            SPDLOG_ERROR("Failed to open encoded file, errno={}", errno);
+        } else {
+            SPDLOG_ERROR("Failed to open encoded file, error_code={}", error_code);
+        }
+        return false;
+    }
+
+    // Generate output directory
+    error_code = create_directory_structure(output_dir, 0700);
+    if (ErrorCode_Success != error_code) {
+        SPDLOG_ERROR(
+                "Failed to create directory structure {}, errno={}",
+                output_dir.c_str(),
+                errno
+        );
+        return false;
+    }
+
+    if (temp_output_dir != output_dir) {
+        // Generate temporary output directory
+        error_code = create_directory_structure(temp_output_dir, 0700);
+        if (ErrorCode_Success != error_code) {
+            SPDLOG_ERROR(
+                    "Failed to create directory structure {}, errno={}",
+                    output_dir.c_str(),
+                    errno
+            );
+            return false;
+        }
+    }
+
+    boost::filesystem::path temp_ir_path{temp_output_dir};
+    temp_ir_path /= m_encoded_file.get_id_as_string() + ".temp." + ir::get_ir_extension_name();
```

Comment:
Do you think the extension name ".clp.zst" makes more sense?

Reply:
you mean remove the ".temp."? sure